OpenBSD 6.3 build: test/arith_uint256_tests.cpp complains about missing argument

<!-- This issue tracker is only for technical issues related to Bitcoin Core.

<!-- Describe the issue -->

When running `gmake check`, test/arith_uint256_tests.cpp complains about missing argument.

<!--- What behavior did you expect? -->

Expected successful `gmake check` results.

<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->
Output from bitcoin/src/test/arith_uint256_tests.cpp.log:

```
Missing an argument value for the parameter run_test in the argument 

Parameter: run_test
 Filters, which test units to include or exclude from test module execution.
 Command line formats:
   --run_test=<test unit filter>
   -t <test unit filter>
 Environment variable: BOOST_TEST_RUN_FILTERS

For detailed help on Boost.Test parameters use:
  test_bitcoin --help
or
  test_bitcoin --help=<parameter name>
```

<!--- How reliably can you reproduce the issue, what are the steps to do so? -->
This issue occurs when running `gmake check`.


<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->

This issue occurs with bitcoin master version 610f4dd7

<!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->

This issue occurs with OpenBSD 6.3 AMD64. 
File [config.log](https://github.com/bitcoin/bitcoin/files/2045192/config.log) attached.

<!-- Any extra information that might be useful in the debugging process. -->
<!--- This is normally the contents of a `debug.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred. -->

I'm able to reproduce this issue under stock OpenBSD 6.3 using the default shell:

```
$ uname -r
6.3
$ echo $0
-ksh
$ gmake check
Making check in src
gmake[1]: Entering directory '/.../bitcoin/src'
gmake[2]: Entering directory '/.../bitcoin/src'
gmake[3]: Entering directory '/.../bitcoin'
gmake[3]: Leaving directory '/.../bitcoin'
gmake  check-TESTS check-local
gmake[3]: Entering directory '/.../bitcoin/src'
gmake[4]: Entering directory '/.../bitcoin/src'
============================================================================
Testsuite summary for Bitcoin Core 0.16.99
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
gmake[4]: Leaving directory '/.../bitcoin/src'
Running tests: from test/arith_uint256_tests.cpp
Missing an argument value for the parameter run_test in the argument

Parameter: run_test
 Filters, which test units to include or exclude from test module execution.
 Command line formats:
   --run_test=<test unit filter>
   -t <test unit filter>
 Environment variable: BOOST_TEST_RUN_FILTERS

For detailed help on Boost.Test parameters use:
  test_bitcoin --help
or
  test_bitcoin --help=<parameter name>
gmake[3]: *** [Makefile:10247: test/arith_uint256_tests.cpp.test] Error 1
gmake[3]: Leaving directory '/.../bitcoin/src'
gmake[2]: *** [Makefile:9996: check-am] Error 2
gmake[2]: Leaving directory '/.../bitcoin/src'
gmake[1]: *** [Makefile:9684: check-recursive] Error 1
gmake[1]: Leaving directory '/.../bitcoin/src'
gmake: *** [Makefile:754: check-recursive] Error 1
```

--- a/src/Makefile.test.include
+++ b/src/Makefile.test.include
@@ -166,8 +166,8 @@ if EMBEDDED_UNIVALUE
 endif
 
 %.cpp.test: %.cpp
-	@echo Running tests: `cat $< | grep "BOOST_FIXTURE_TEST_SUITE(\|BOOST_AUTO_TEST_SUITE(" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1` from $<
-	$(AM_V_at)$(TEST_BINARY) -l test_suite -t "`cat $< | grep "BOOST_FIXTURE_TEST_SUITE(\|BOOST_AUTO_TEST_SUITE(" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" > $<.log 2>&1 || (cat $<.log && false)
+	@echo Running tests: `cat $< | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1` from $<
+	$(AM_V_at)$(TEST_BINARY) -l test_suite -t "`cat $< | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" > $<.log 2>&1 || (cat $<.log && false)
 
 %.json.h: %.json
 	@$(MKDIR_P) $(@D)

